cpp/dolfinx/common/IndexMap.cpp

```diff
@@ -694,9 +695,10 @@ void IndexMap::scatter_fwd(tcb::span<const T> local_data,
 
   // Copy into sending buffer
   std::vector<T> data_to_send(displs_send.back());
-  for (std::size_t i = 0; i < _shared_indices.size(); ++i)
+  const std::vector<std::int32_t>& indices = _shared_indices->array();
+  for (std::size_t i = 0; i < indices.size(); ++i)
   {
-    const int index = _shared_indices[i];
+    const int index = indices[i];
```

Comment:
Let's be explicit here with `std::int32_t` vs `std::int64_t` and in other places where there could be confusion.

Reply:
After checking `IndexMap`, we now have:
Fixed-width integers used for differentiating local and global indices and quantities.  Only a few places had minor inconsistencies. 
And `int`s used for block size, ranks and number of neighbors, and size of data to send/receive.